gqltesting/testing.go

```diff
@@ -43,17 +43,14 @@ func RunTest(t *testing.T, test *Test) {
 		test.Context = context.Background()
 	}
 	result := test.Schema.Exec(test.Context, test.Query, test.OperationName, test.Variables)
-	if result.Errors != nil {
-		checkErrors(t, test.ExpectedErrors, result.Errors)
-		return
-	}
+	checkErrors(t, test.ExpectedErrors, result.Errors)
 	// Verify JSON to avoid red herring errors.
 	got, err := formatJSON(result.Data)
-	if err != nil {
+	if err != nil && test.ExpectedResult != "" {
 		t.Fatalf("got: invalid JSON: %s\n response value is: [%s]", err, got)
 	}
 	want, err := formatJSON([]byte(test.ExpectedResult))
-	if err != nil {
+	if err != nil && test.ExpectedResult != ""  {
```

Comment:
Why is this required? And why do you skip it if the expected result is `""`?

Reply:
After a second look, I understood what's going on. In case of an error, fail only if there is expected result.

Reply:
Yes, because if an error happens and returns empty response it will give error when doing  `formatJSON` and this will fail the test even if you are expecting the error and an the empty result.